tests/org.pitest.pitclipse.ui.tests/src/org/pitest/pitclipse/ui/behaviours/pageobjects/PitSummaryView.java

```diff
@@ -16,34 +16,86 @@
+import org.eclipse.swtbot.eclipse.finder.widgets.SWTBotView;
+import org.eclipse.swtbot.swt.finder.finders.UIThreadRunnable;
+import org.eclipse.swtbot.swt.finder.results.VoidResult;
+import static org.pitest.pitclipse.ui.view.PitView.BACK_BUTTON_TEXT;
+import static org.pitest.pitclipse.ui.view.PitView.FORWARD_BUTTON_TEXT;
+import static org.pitest.pitclipse.ui.view.PitView.HOME_BUTTON_TEXT;
 
 public class PitSummaryView {
+    private static String SUMMARY_VIEW_TITLE = "PIT Summary";
+    private SWTWorkbenchBot bot;
+    private SWTBotView summaryView;
 
-    private PitResultsView lastResults = null;
-
-    public PitSummaryView() {
+    public PitSummaryView(SWTWorkbenchBot bot) {
+        this.bot = bot;
     }
 
-    public void waitForUpdate() {
-        try {
-            lastResults = PitNotifier.INSTANCE.getResults();
-        } catch (InterruptedException e) {
-            throw new StepException(e);
+    public void showViewIfNotOpen() {
+        if (summaryView == null) {
+            summaryView = bot.viewByTitle(SUMMARY_VIEW_TITLE);
         }
+        summaryView.show();
+    }
+
+    public void clickButtonWithText(String text) {
+        showViewIfNotOpen();
+        bot.toolbarButton(text).click();
+    }
+
+    public String getCurrentBrowserUrl() {
+        showViewIfNotOpen();
+        return summaryView.bot().browser().getUrl();
     }
 
-    public int getClassesTested() {
-        return lastResults.getClassesTested();
+    public String clickBack() {
+        clickButtonWithText(BACK_BUTTON_TEXT);
+        summaryView.bot().browser().waitForPageLoaded();
+        return getCurrentBrowserUrl();
     }
 
-    public double getOverallCoverage() {
-        return lastResults.getTotalCoverage();
+    public String clickHome() {
+        clickButtonWithText(HOME_BUTTON_TEXT);
+        summaryView.bot().browser().waitForPageLoaded();
+        return getCurrentBrowserUrl();
     }
 
-    public double getMutationCoverage() {
-        return lastResults.getMutationCoverage();
+    public String clickForward() {
+        clickButtonWithText(FORWARD_BUTTON_TEXT);
+        summaryView.bot().browser().waitForPageLoaded();
+        return getCurrentBrowserUrl();
+    }
+
+    /**
+     * Sets the link in the browser via javaScript, because bot can't click links in
+     * browser
+     * @param hyperLinkText link which to set as url, can be relative
+     * @return new url of browser
+     */
+    public String setLink(String hyperLinkText) {
+        showViewIfNotOpen();
+        // use this instead of the SWTBrowser.execute(), because we need syncExec
+        final String link = "window.location.href = '" + hyperLinkText + "'";
+        summaryView.bot().browser().waitForPageLoaded();
+        UIThreadRunnable.syncExec(new VoidResult() {
+            @Override
+            public void run() {
+                summaryView.bot().browser().widget.execute(link);
+            }
+        });
```

Comment:
Use a lambda expression instead of an anonymous class when possible:
```java
UIThreadRunnable.syncExec(() ->
        summaryView.bot().browser().widget.execute(link)
);
```